cartoframes/data/observatory/repository/category_repo.py

```diff
@@ -0,0 +1,41 @@
+from cartoframes.data.observatory.repository.repo_client import RepoClient
+
+
+def get_category_repo():
+    return CategoryRepository()
+
+
+class CategoryRepository(object):
+
+    def __init__(self):
+        self.client = RepoClient()
+
+    def get_all(self):
+        return self._to_categories(self.client.get_categories())
+
+    def get_by_id(self, category_id):
+        result = self.client.get_categories('id', category_id)
+
+        if len(result) == 0:
+            return None
+
+        return self._to_category(result[0])
+
+    def get_by_country(self, iso_code):
+        # TODO
+        pass
+
+    @staticmethod
+    def _to_category(result):
+        from cartoframes.data.observatory.category import Category
```

Comment:
Ok, so here it comes a nasty one. Given the current requirements for the API there's a cyclic dependency I couldn't avoid, since the entity classes (Dataset, Category, ...) are also responsible for their one retrieval from the server (so they have a dependency on these repository classes) and at the same time they're responsible for other entities' retrieval (like Dataset needs to know how retrieve Variables) which creates inter-dependencies among the entities.

I came up with this solution but it may be a better and less ugly one for sure :D

Reply:
I'd go with the solution that works at the moment.